packages/app-root/src/app/style-registry.js

```diff
@@ -0,0 +1,27 @@
+'use client'
+
+import { useServerInsertedHTML } from 'next/navigation'
+import { ServerStyleSheet, StyleSheetManager } from 'styled-components'
+import { useState } from 'react'
+
+export default function StyledComponentsRegistry({
+  children
+}) {
+  // Only create stylesheet once with lazy initial state
+  // x-ref: https://reactjs.org/docs/hooks-reference.html#lazy-initial-state
+  const [styledComponentsStyleSheet] = useState(() => new ServerStyleSheet())
+
+  useServerInsertedHTML(() => {
```

Comment:
This hook name is confusing in a client-side component.

Reply:
Although this is a client component, you noted earlier the downloaded HTML includes an inline stylesheet. Client components still run once on the server, and then again in the browser. This code is copied from the styled-components example: https://nextjs.org/docs/app/building-your-application/styling/css-in-js#styled-components

Do you recommend changing the function name?